### PR TITLE
feat: scaffold out triggers and implement ping

### PR DESCRIPTION
**Pull request by Lauraducky**

### CODE REVIEW

File: src/PockyBot.NET/Services/Triggers/Ping.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        protected override bool DirectMessageAllowed => true;
+
+        protected override bool CanHaveArgs => false;
+
+        protected override bool PermissionsRequired => false;
+
+        public override Message Respond(Message message) {
+            return new Message {
+                Text = "pong. I'm alive!"
```
Comment (arthurs) [inline src/PockyBot.NET/Services/Triggers/Ping.cs]: shouldn't this have a version number in the response

Comment (Lauraducky) [reply src/PockyBot.NET/Services/Triggers/Ping.cs]: Which version are we talking? The library version? A version provided to the bot via environment variable? Perhaps we can add that later? I'm trying to keep this PR stripped down so it's easier to review

File: src/PockyBot.NET/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using PockyBot.NET.Configuration;
+using PockyBot.NET.Persistence;
+using PockyBot.NET.Services;
+using PockyBot.NET.Services.Triggers;
+
+namespace PockyBot.NET
```
Comment (Chromadream) [inline src/PockyBot.NET/Extensions.cs]: I'd rather put extensions as its own namespace, and make these a `Configuration` static class or something.

**Comment by Evangelinexx** [comment]
What are your thoughts on using something like DBUp to stand up the database and update it automatically? Or separate PR?

File: database/populate_roles.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO roles (userid, role)
+VALUES
+('{id of an admin}', 'ADMIN'),
```
Comment (jimcullenaus) [inline database/populate_roles.sql]: ends with a comma?

File: src/PockyBot.NET/Services/TriggerResponseTester.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            else
+            {
+                return message.Text.Trim().ToLower() == trigger.Command.ToLower();
+            }
+        }
+
+        private bool HasPermission(string senderId, string[] permissions)
+        {
+            var user = _pockyUserRepository.GetUser(senderId);
+            return user != null && user.Roles.Any(x => permissions.Contains(x.UserRole));
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/TriggerResponseTester.cs]: case insensitive string searches

File: src/PockyBot.NET/Services/TriggerResponseTester.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                return false;
+            }
+
+            if (trigger.CanHaveArgs)
+            {
+                return message.MessageParts[1].Text.Trim().ToLower().StartsWith(trigger.Command.ToLower());
+            }
+            else
+            {
+                return message.MessageParts.Length == 2 && message.MessageParts[1].Text.Trim().ToLower() == trigger.Command.ToLower();
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/TriggerResponseTester.cs]: shouldn't need `ToLower`. Use `string.Equals(one, two, StringComparison.InvariantCultureIgnoreCase)`.

File: src/PockyBot.NET/Extensions.cs (hunk 2 above)
Comment (Lauraducky) [reply src/PockyBot.NET/Extensions.cs]: I disagree. From the perspective of someone using this library, I think the extension methods are core functionality. Adding another namespace will just pollute things. So as it is currently, someone will be able to use it with:

```
using PockyBot.NET;

services.ConfigurePockyBot(Configuration);
```

I'd prefer that most of the core functionality (which is pretty much everything that is declared public, note that a lot of the classes in this pr are internal) be living under the root namespace. This is a pretty standard way of writing libraries. Hopefully that makes sense.

Comment (Chromadream) [reply src/PockyBot.NET/Extensions.cs]: That actually sounds sensible. Didn't really think of it that way.

File: src/PockyBot.NET/Services/Triggers/Ping.cs (hunk 1 above)
Comment (arthurs) [reply src/PockyBot.NET/Services/Triggers/Ping.cs]: I would have it pull an environment variable that it sets, but that can be overwritten by the project that implements the library? but yeah, maybe add in a later PR

**Pull request merged**